apps/omg_child_chain/lib/omg_child_chain/fee_server.ex

```diff
@@ -28,28 +28,20 @@ defmodule OMG.ChildChain.FeeServer do
   alias OMG.Fees
   alias OMG.Status.Alert.Alarm
 
-  @fee_file_check_interval_ms Application.fetch_env!(:omg_child_chain, :fee_file_check_interval_ms)
-  @fee_buffer_duration_ms Application.fetch_env!(:omg_child_chain, :fee_buffer_duration_ms)
-
   def start_link(opts) do
     GenServer.start_link(__MODULE__, opts, name: __MODULE__)
   end
 
   def init(args) do
     :ok = ensure_ets_init()
 
-    # This will crash the process by returning {:error, :fee_adapter_not_defined}
-    # if the fee_adapter is not set in the config
-    {:ok, fee_adapter} = get_fee_adapter()
-
     {:ok, state} =
       args
       |> Enum.into(%{})
```

Comment:
is this not using a struct as it's internal process state? why not?

Reply:
Maybe it's not worth to as it basically gets all values from the configuration / process init's args. WDYT?

Reply:
it's easier to reason about it and read code. if you know the structure of the state, rather then being just dynamically "created". it also follows the convention of other genservers we have (I think! there might be some unfortunate misses, but it's a good habit).